frontend/packages/dev-console/src/actions/add-resources.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import {
+  GitAltIcon,
+  OsImageIcon,
+  CatalogIcon,
+  CubeIcon,
+  DatabaseIcon,
+  UnknownIcon,
+} from '@patternfly/react-icons';
+import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { KebabOption } from '@console/internal/components/utils';
+import { ImportOptions } from '../components/import/import-types';
+
+const getAddPageUrl = (obj: K8sResourceKind, type: string): string => {
+  let pageUrl = '';
+  const {
+    metadata: { namespace: ns },
+  } = obj;
+  switch (type) {
+    case ImportOptions.GIT:
+      pageUrl = `/import/ns/${ns}?importType=git`;
+      break;
+    case ImportOptions.CONTAINER:
+      pageUrl = `/deploy-image/ns/${ns}?preselected-ns=${ns}`;
+      break;
+    case ImportOptions.CATALOG:
+      pageUrl = '/catalog';
+      break;
+    case ImportOptions.DOCKERFILE:
+      pageUrl = `/import/ns/${ns}?importType=docker`;
+      break;
+    case ImportOptions.DATABASE:
+      pageUrl = `/catalog/ns/${ns}?category=databases`;
+      break;
+    default:
+      pageUrl = `add/ns/${ns}`;
+  }
+  const seperator = pageUrl.includes('?') ? '&' : '?';
+  return `${pageUrl}${seperator}component=${obj.metadata.name}&isKnativeDisabled=${obj.kind ===
```

Comment:
It's important to note URLs are tricky beasts... always use `encodeURIComponent` when crafting a url with dynamic values ([MDN ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent))

```suggestion
  return `${pageUrl}${seperator}component=${encodeURIComponent(obj.metadata.name)}&isKnativeDisabled=${obj.kind ===
```